python/samples/concepts/memory/azure_ai_search_hotel_sample.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import asyncio
+from typing import Annotated, Any, TypeVar
+
+from pydantic import BaseModel
+
+from semantic_kernel import Kernel
+from semantic_kernel.connectors.ai.open_ai import OpenAIEmbeddingPromptExecutionSettings, OpenAITextEmbedding
+from semantic_kernel.connectors.memory.azure_ai_search import AzureAISearchCollection
+from semantic_kernel.data import (
+    VectorSearchFilter,
+    VectorSearchOptions,
+    VectorStoreRecordDataField,
+    VectorStoreRecordKeyField,
+    VectorStoreRecordUtils,
+    VectorStoreRecordVectorField,
+    vectorstoremodel,
+)
+
+
+@vectorstoremodel
```

Comment:
Is this file demonstrating a new concept or is more of a Demo? If it's the latter, should it be moved into the Demos folder?